pkg/cmdutil/json_flags.go

```diff
@@ -74,7 +74,13 @@ func AddJSONFlags(cmd *cobra.Command, exportTarget *Exporter, fields []string) {
 			sort.Strings(fields)
 			return JSONFlagError{fmt.Errorf("Specify one or more comma-separated fields for `--json`:\n  %s", strings.Join(fields, "\n  "))}
 		}
-		return c.Parent().FlagErrorFunc()(c, e)
+		if c.Parent() != nil {
+			return c.Parent().FlagErrorFunc()(c, e)
+		}
+		if e == pflag.ErrHelp {
+			return e
+		}
+		return e
 	})
```

Comment:
When building `gh`, all subcommands inherits the `root`'s FlagErrorFunc declared here:
https://github.com/cli/cli/blob/640a089e55f28af5386bfb020ce6fcd77fb300ef/pkg/cmd/root/root.go#L65

But when testing a specific cmd isolated there's no `Parent` cmd so it panics when doing an error-check test like this:
https://github.com/cli/cli/runs/2876079048#step:5:94
(You can see it by adding an error-check test to any command that exports JSON like `repo list`, `pr view`, etc).

I've added this check to avoid accessing `c.Parent` if it doesn't exists, though I'm not sure if there's any better solution for this.